server/database/sql/create_functions.sql

```diff
@@ -5,3 +5,103 @@ begin
 return NEW;
 end;
 $$ language plpgsql;
+
+CREATE OR REPLACE FUNCTION public.trigger_insert_words()
+    RETURNS trigger
+    LANGUAGE 'plpgsql'
+    COST 100
+    VOLATILE NOT LEAKPROOF
+AS $BODY$
+begin
+    insert into word(word, tracker) select
+		unnest(
+			array_remove(
+				regexp_split_to_array(
+					NEW.overview,
+					'(?!'')[[:punct:]\s*]'
+				),
+				''
+			)
+		) as word,
+		new.id as tracker;
+return NEW;
+end;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION public.trigger_delete_words()
+    RETURNS trigger
+    LANGUAGE 'plpgsql'
+    COST 100
+    VOLATILE NOT LEAKPROOF
+AS $BODY$
+begin
+    delete from word where tracker = new.id;
```

Comment:
@ChrisShep98 this is a function that deletes all the words related to today from the word table. we need this for when the user updates their submission. in order to avoid duplicates, we delete all the words that have todays tracker table foreign key, then rerun the insert function for the updated submission